hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/ContainerDataScanner.java

```diff
@@ -95,14 +97,19 @@ public void runIteration() {
     while (!stopping && itr.hasNext()) {
       Container c = itr.next();
       if (c.shouldScanData()) {
+        ContainerData containerData = c.getContainerData();
+        long containerId = containerData.getContainerID();
         try {
+          logScanStart(containerData);
           if (!c.scanData(throttler, canceler)) {
             metrics.incNumUnHealthyContainers();
-            controller.markContainerUnhealthy(
-                c.getContainerData().getContainerID());
+            controller.markContainerUnhealthy(containerId);
```

Comment:
We should also call `logScanCompleted` and `updateDataScanTimestamp` in the failure path.

Reply:
I would avoid this for two reasons:

1. The full scan includes a scan of the metadata, too, and the failure may be due to metadata problem.  Eg. if the `.container` file is missing or invalid etc.  In that case we cannot update the timestamp in the file.
2. Unhealthy containers are skipped during further iterations, so the timestamp would not make much difference anyway.